lib/src/simulator.dart

```diff
@@ -265,6 +271,12 @@ class Simulator {
     _simulationEndRequested = true;
   }
 
+  /// End Simulation and return the exception
+  static void throwException(Exception err, StackTrace stacktrace) {
+    _exception = err;
```

Comment:
As implemented, this only records the last `throwException` before the simulator finally raises the exception.  You probably either want to save the *first* exeption, or all of them.  If we capture just the first, then we should think what to do if there's more.

Reply:
I changed to List type. Not sure it will help.